{"primaryContentSections":[{"kind":"declarations","declarations":[{"tokens":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"externalParam","text":"_"},{"kind":"text","text":" "},{"kind":"internalParam","text":"challenge"},{"kind":"text","text":": "},{"kind":"typeIdentifier","identifier":"doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS\/CustomAuthFinishAuthenticationChallenge","preciseIdentifier":"s:13OneginiSDKiOS39CustomAuthFinishAuthenticationChallengeP","text":"CustomAuthFinishAuthenticationChallenge"},{"kind":"text","text":", "},{"kind":"externalParam","text":"underlyingError"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?)"}],"languages":["swift"],"platforms":["iOS"]}]},{"kind":"parameters","parameters":[{"name":"challenge","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"challenge for which response is made"}]}]},{"name":"underlyingError","content":[{"type":"paragraph","inlineContent":[{"type":"text","text":"error specifying the reason behind the cancellation."},{"type":"text","text":" "},{"type":"text","text":"It will ba added as a underlying error within the cancellation error returned in the authentication failure method."}]}]}]}],"schemaVersion":{"major":0,"minor":3,"patch":0},"sections":[],"variants":[{"paths":["\/documentation\/oneginisdkios\/customauthfinishauthenticationchallengesender\/cancel(_:underlyingerror:)"],"traits":[{"interfaceLanguage":"swift"}]}],"identifier":{"url":"doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS\/CustomAuthFinishAuthenticationChallengeSender\/cancel(_:underlyingError:)","interfaceLanguage":"swift"},"abstract":[{"type":"text","text":"Method to cancel challenge."}],"kind":"symbol","metadata":{"role":"symbol","title":"cancel(_:underlyingError:)","roleHeading":"Instance Method","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"CustomAuthFinishAuthenticationChallenge","preciseIdentifier":"s:13OneginiSDKiOS39CustomAuthFinishAuthenticationChallengeP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"underlyingError"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?)"}],"symbolKind":"method","externalID":"s:13OneginiSDKiOS45CustomAuthFinishAuthenticationChallengeSenderP6cancel_15underlyingErroryAA0defgH0_p_s0L0_pSgtF","required":true,"modules":[{"name":"OneginiSDKiOS"}]},"hierarchy":{"paths":[["doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS","doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS\/CustomAuthFinishAuthenticationChallengeSender"]]},"references":{"doc://com.onegini.OneginiSDKiOS/documentation/OneginiSDKiOS":{"role":"collection","title":"OneginiSDKiOS","abstract":[],"identifier":"doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS","kind":"symbol","type":"topic","url":"\/documentation\/oneginisdkios"},"doc://com.onegini.OneginiSDKiOS/documentation/OneginiSDKiOS/CustomAuthFinishAuthenticationChallengeSender/cancel(_:underlyingError:)":{"role":"symbol","title":"cancel(_:underlyingError:)","fragments":[{"kind":"keyword","text":"func"},{"kind":"text","text":" "},{"kind":"identifier","text":"cancel"},{"kind":"text","text":"("},{"kind":"typeIdentifier","text":"CustomAuthFinishAuthenticationChallenge","preciseIdentifier":"s:13OneginiSDKiOS39CustomAuthFinishAuthenticationChallengeP"},{"kind":"text","text":", "},{"kind":"externalParam","text":"underlyingError"},{"kind":"text","text":": "},{"kind":"typeIdentifier","text":"Error","preciseIdentifier":"s:s5ErrorP"},{"kind":"text","text":"?)"}],"abstract":[{"type":"text","text":"Method to cancel challenge."}],"identifier":"doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS\/CustomAuthFinishAuthenticationChallengeSender\/cancel(_:underlyingError:)","kind":"symbol","required":true,"type":"topic","url":"\/documentation\/oneginisdkios\/customauthfinishauthenticationchallengesender\/cancel(_:underlyingerror:)"},"doc://com.onegini.OneginiSDKiOS/documentation/OneginiSDKiOS/CustomAuthFinishAuthenticationChallengeSender":{"role":"symbol","title":"CustomAuthFinishAuthenticationChallengeSender","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"CustomAuthFinishAuthenticationChallengeSender"}],"abstract":[{"type":"text","text":"Protocol describing an SDK object waiting for the response to the authentication with Custom Authenticator."}],"identifier":"doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS\/CustomAuthFinishAuthenticationChallengeSender","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"CustomAuthFinishAuthenticationChallengeSender"}],"url":"\/documentation\/oneginisdkios\/customauthfinishauthenticationchallengesender"},"doc://com.onegini.OneginiSDKiOS/documentation/OneginiSDKiOS/CustomAuthFinishAuthenticationChallenge":{"role":"symbol","title":"CustomAuthFinishAuthenticationChallenge","fragments":[{"kind":"keyword","text":"protocol"},{"kind":"text","text":" "},{"kind":"identifier","text":"CustomAuthFinishAuthenticationChallenge"}],"abstract":[{"type":"text","text":"Represents authentication with custom authenticator challenge. It provides all information about the challenge and a sender awaiting for a response."}],"identifier":"doc:\/\/com.onegini.OneginiSDKiOS\/documentation\/OneginiSDKiOS\/CustomAuthFinishAuthenticationChallenge","kind":"symbol","type":"topic","navigatorTitle":[{"kind":"identifier","text":"CustomAuthFinishAuthenticationChallenge"}],"url":"\/documentation\/oneginisdkios\/customauthfinishauthenticationchallenge"}}}